underscore.js

```diff
@@ -779,13 +779,20 @@
 
   // Memoize an expensive function by storing its results.
   _.memoize = function(func, hasher) {
+    function getCache() {
+      if (Object.create) {
+        return Object.create(null);
+      }
+      return {};
+    }
+
     var memoize = function(key) {
       var cache = memoize.cache;
       var address = '' + (hasher ? hasher.apply(this, arguments) : key);
-      if (!_.has(cache, address)) cache[address] = func.apply(this, arguments);
+      if (typeof cache[address] === 'undefined') cache[address] = func.apply(this, arguments);
```

Comment:
Wouldn't this need to be forked a bit more for enviros without `Object.create` which still need the own property checks.